### ISSUE TITLE
iOS - Supporting Xcode TestPlans

### ISSUE THREAD

**Issue by abhishekbedi1432** [MarathonLabs/marathon#688]
### **Is your feature request related to a problem? Please describe.**
Running your tests with different environments, localizations and sanitizers is a good way to catch more bugs. You can use schemes to create the different configurations but that quickly becomes unmanageable.

Apple added test plans to Xcode 11 to make it easier to rerun tests with varying test configurations. Here’s my notes on how to get started with test plans.

What is a Test Plan?
An Xcode test plan provides a way to run a selections of tests with various test configurations. The test plan is a JSON file with the .xctestplan extension that you add to your Xcode project and reference from a scheme. It has three main elements:

Test targets: One or more test targets (unit or UI). You don’t have to run all tests in the test target. For each test plan, you select the tests the plan will run and whether the tests can run in parallel.

Default options: A set of default (shared) settings that can be overridden by specific configurations. These are the settings you would normally find in the scheme editor: launch arguments, localization settings, screenshot settings, text execution (alphabetic or random), runtime sanitizers, thread checker, and malloc guards.

Configurations: One or more configurations that override the default options. A test plan runs the selected tests multiple times, once for each of the configurations

### **Describe the solution you'd like**
Supporting building the application both with & without test plans

**Comment by Malinskiy** [comment]
Duplicate of #620
